packages/material-ui-lab/src/DateRangePickerDay/DateRangePickerDay.test.tsx

```diff
@@ -48,12 +32,21 @@ describe('<DateRangePickerDay />', () => {
     () => ({
       classes,
       inheritComponent: 'button',
+      muiName: 'MuiDateRangePickerDay',
       render: (node: React.ReactNode) =>
         render(<LocalizationProvider dateAdapter={AdapterClassToUse}>{node}</LocalizationProvider>),
       mount: localizedMount,
       refInstanceof: window.HTMLButtonElement,
       // cannot test reactTestRenderer because of required context
-      skip: ['componentProp', 'reactTestRenderer', 'propsSpread', 'refForwarding'],
+      skip: [
+        'componentProp',
+        'componentsProp',
+        'reactTestRenderer',
+        'propsSpread',
+        'refForwarding',
+        'themeDefaultProps',
```

Comment:
We should either support all: theme props, theme variants & theme overrides, or none. Why is it ok to not have the default props & variants, but it's ok to have ovrerides? Was implemented like that before?

Reply:
The implementation before pass `{...other}` to nested child (`PickersDay`) which is not work with `themeDefaultProps` test (the test will check the firstChild).

```jsx
const {
    className,
    // ...
    ...other
  } = props;
  
  <DateRangePickerDayRoot>
    <DateRangePickerDayRangeIntervalPreview>
      <DateRangePickerDayDay {...other}>
```

Reply:
> We should either support all: theme props, theme variants & theme overrides, or none.

I would add one exception (does not apply here). Let's say you have A as a public component and B as a private component, used by A. To me, it seems to make a lot of sense to have A support theme props, theme variants and theme overrides, and B only support theme overrides, as is it was part of A (using the same muiName) 

> The implementation before pass {...other} to nested child (PickersDay)

Maybe we could leave a TODO as it breaks the convention.